x/evm/types/command.go

```diff
@@ -138,6 +138,25 @@ func NewApproveContractCallCommand(
 	}
 }
 
+// NewApproveContractCallCommandFromGeneralMessage creates a command to approve contract call
+func NewApproveContractCallCommandFromGeneralMessage(
+	chainID sdk.Int,
+	keyID multisig.KeyID,
+	msg nexus.GeneralMessage,
+) Command {
+	contractAddress := common.HexToAddress(msg.Receiver)
```

Comment:
`common.HexToAddress` returns address(0) when `msg.Receiver` is not a valid Ethereum address. This should be dealt with somewhere. Preferably, the `.ValidateBasic` funtion of the `CallContractRequest` should prevent this from happening.

Reply:
added a `ValidateAddress` method to nexus keeper